File: frontend/ReadBuddy/ReadBuddy/Models/Quiz/Answer.cs
// ReadBuddy\Models\Quiz\Answer.cs
using ReadBuddy.Models.TTS;
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace ReadBuddy.Models.Quiz
{
    public class Answer
    {
        [JsonPropertyName("id")]
        public string Id { get; private set; }

        [JsonPropertyName("user_id")]
        public string UserId { get; private set; }

        [JsonPropertyName("content")]
        public string Content { get; set; }

        [JsonPropertyName("audio_zip_urls")]
        public List<string> AudioZipUrls { get; private set; }

        public TTSResult? ttsResult { get; set; } = null;

        public Answer(string userId, string content, List<string> audioZipUrls, string id)
        {
            UserId = userId;
            Id = id;
            Content = content;
            AudioZipUrls = audioZipUrls;
        }
    }
}

File: backend/app/models/answer.py
# app/models/answer.py

from typing import List
import uuid

class Answer:
    """
    Represents an answer in a quiz, containing the user's response and associated audio files.
    """
    def __init__(self, user_id: str, content: str, audio_zip_urls: List[str] = None, id :str = None):
        self.user_id = user_id
        self.id = f"a-{uuid.uuid4().hex[:12]}" if not id else id
        self.content = content
        self.audio_zip_urls = audio_zip_urls if audio_zip_urls else []

    def to_dict(self) -> dict:
        return {
            "id": self.id,
            "user_id": self.user_id,
            "content": self.content,
            "audio_zip_urls": self.audio_zip_urls
        }